src/clipping/sutherlandhodgman.jl

```diff
@@ -14,6 +14,51 @@ The Sutherland-Hodgman algorithm for clipping polygons.
 """
 struct SutherlandHodgman <: ClippingMethod end
 
-function clip(poly::Polygon, other::Geometry, ::SutherlandHodgman)
-  @error "not implemented"
+function clip(poly::Polygon, other::Geometry, method::ClippingMethod) 
+  r = [clip(ring, boundary(other), method) for ring in rings(poly)]
+  r = filter(!isnothing, r)
+
+  if isempty(r)
+    nothing
+  elseif hasholes(poly)
+    PolyArea(r[1], r[2:end])
+  else
+    PolyArea(r[1])
+  end
+end
+
+function clip(ring::Ring{Dim,T}, other::Ring{Dim,T}, ::SutherlandHodgman) where {Dim,T}
+  o = vertices(other)
+  n = length(o)
+  v = vertices(ring)
+
+  outside = orientation(other) == :CW ? :LEFT : :RIGHT
```

Comment:
perhaps we should reverse(other) in case the orientation is incorrect, it is better than changing the algorithm's constants.